src/utils/powersystems_utils.jl

```diff
@@ -1,28 +1,76 @@
 function get_available_components(
-    ::Type{T},
+    model::DeviceModel{T, <:AbstractDeviceFormulation},
     sys::PSY.System,
     ::Nothing = nothing,
 ) where {T <: PSY.Component}
-    return PSY.get_components(PSY.get_available, T, sys)
+    subsystem = get_subsystem(model)
+    return PSY.get_components(PSY.get_available, T, sys; subsystem_name = subsystem)
 end
 
 function get_available_components(
-    ::Type{T},
+    model::DeviceModel{T, <:AbstractDeviceFormulation},
+    sys::PSY.System,
+    f::Function,
```

Comment:
Does these functions need to reorder the parameters so that filter_function is first?

Reply:
We can incorporate that change for sure. It is a legacy from old code.